### PR TITLE
Move generated adapter to companion object

### PR DESCRIPTION
**Pull request by mc-lovin**
Fixes https://github.com/square/wire/issues/707

### CODE REVIEW

File: wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt [hunk 1]
```diff
@@ -22,42 +23,46 @@ data class Person(
...
 ) {
-    object ADAPTER : ProtoAdapter<Person>(FieldEncoding.LENGTH_DELIMITED, Person::class.java) {
-        override fun encodedSize(value: Person): Int = ProtoAdapter.STRING.encodedSizeWithTag(1, value.name) +
-            ProtoAdapter.INT32.encodedSizeWithTag(2, value.id) +
-            ProtoAdapter.STRING.encodedSizeWithTag(3, value.email) +
-            PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phone) +
-            value.unknownFields.size()
+    companion object {
+        @JvmField
+        val ADAPTER: ProtoAdapter<Person> =
```
Comment (mc-lovin) [inline wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt]: @Egorand is there a way we can get rid of the explicit type here if we are initializing the property?

Comment (Egorand) [reply wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt]: No, property types are always declared explicitly.

**Comment by JakeWharton** [comment]
Dupe of #710? Not that it matters...

**Comment by Egorand** [comment]
@JakeWharton no, #710 fixes `Parcelable` implementation.

**Comment by Egorand** [comment]
@mc-lovin I suggest getting rid of the test class containing failing tests (in a separate PR), that kind of tests is not very helpful. #712 sets up proper integration testing that is easier to maintain and provides better feedback on failures.

**Comment by mc-lovin** [comment]
@Egorand PTAL and merge

**Pull request merged**